xwiki-platform-core/xwiki-platform-component/xwiki-platform-component-wiki/src/main/java/org/xwiki/component/wiki/internal/DefaultWikiObjectComponentManagerEventListener.java

```diff
@@ -105,4 +114,30 @@ public void onEvent(Event event, Object source, Object data)
             this.wikiObjectComponentManagerEventListenerProxy.registerAllObjectComponents();
         }
     }
+
+    /**
+     * Get the {@link WikiObjectComponentBuilder} associated to the given {@link DocumentReference}.
+     * This method checks if a component builder exists for the document reference and then for the local document
+     * reference. If no builder can is found, returns null.
+     *
+     * @param xClassReference the class reference used when searching for the component builder
+     * @return the first component builder that has been found, null if no builder is found
+     */
+    private WikiObjectComponentBuilder getAssociatedComponentBuilder(DocumentReference xClassReference)
+    {
+        // First, try to find a component that has a hint related to the DocumentReference of the XClass
+        try {
+            return this.componentManager.getInstance(WikiObjectComponentBuilder.class,
+                    this.entityReferenceSerializer.serialize(xClassReference));
+        } catch (ComponentLookupException e1) {
+            try {
+                // Try to find a component using the LocalDocumentReference of the XClass
+                return this.componentManager.getInstance(WikiObjectComponentBuilder.class,
+                        this.entityReferenceSerializer.serialize(xClassReference.getLocalDocumentReference()));
```

Comment:
You will get better performance by using the "local" serializer, getLocalDocumentReference() create a new LocalDocumentReference when you call it.